multinode_integration_tests/tests/verify_bill_payment.rs

```diff
@@ -65,98 +41,115 @@ fn verify_bill_payment() {
-        exp_final_service_fee_balance_serv_node_3_minor: owed_to_serving_node_3_minor,
+        ),
     };
 
-    let stimulate_payments =
-        |cluster: &mut MASQNodeCluster,
-         real_consuming_node: &MASQRealNode,
-         _global_test_config: &TestInputsOutputsConfig| {
-            for _ in 0..6 {
-                cluster.start_real_node(
-                    NodeStartupConfigBuilder::standard()
-                        .chain(Chain::Dev)
-                        .neighbor(real_consuming_node.node_reference())
-                        .build(),
-                );
-            }
-        };
+    test_body(
+        test_inputs,
+        assertions_values,
+        stimulate_consuming_node_to_pay_for_test_with_sufficient_funds,
+        activating_serving_nodes_for_test_with_sufficient_funds,
+    );
+}
 
-    let start_serving_nodes_and_run_check =
-        |cluster: &mut MASQNodeCluster,
-         serving_node_1_attributes: ServingNodeAttributes,
-         serving_node_2_attributes: ServingNodeAttributes,
-         serving_node_3_attributes: ServingNodeAttributes,
-         _test_global_config: &TestInputsOutputsConfig| {
-            let serving_node_1 = cluster.start_named_real_node(
-                &serving_node_1_attributes.name,
-                serving_node_1_attributes.index,
-                serving_node_1_attributes.config,
-            );
-            let serving_node_2 = cluster.start_named_real_node(
-                &serving_node_2_attributes.name,
-                serving_node_2_attributes.index,
-                serving_node_2_attributes.config,
-            );
-            let serving_node_3 = cluster.start_named_real_node(
-                &serving_node_3_attributes.name,
-                serving_node_3_attributes.index,
-                serving_node_3_attributes.config,
-            );
-            for _ in 0..6 {
-                cluster.start_real_node(
-                    NodeStartupConfigBuilder::standard()
-                        .chain(Chain::Dev)
-                        .neighbor(serving_node_1.node_reference())
-                        .neighbor(serving_node_2.node_reference())
-                        .neighbor(serving_node_3.node_reference())
-                        .build(),
-                );
-            }
+fn stimulate_consuming_node_to_pay_for_test_with_sufficient_funds(
+    cluster: &mut MASQNodeCluster,
+    real_consuming_node: &MASQRealNode,
+    _wholesome_config: &WholesomeConfig,
+) {
+    for _ in 0..6 {
+        cluster.start_real_node(
+            NodeStartupConfigBuilder::standard()
+                .chain(Chain::Dev)
+                .neighbor(real_consuming_node.node_reference())
+                .build(),
+        );
+    }
+}
 
-            (serving_node_1, serving_node_2, serving_node_3)
-        };
+fn activating_serving_nodes_for_test_with_sufficient_funds(
+    cluster: &mut MASQNodeCluster,
+    wholesome_values: &WholesomeConfig,
+) -> [MASQRealNode; 3] {
```

Comment:
It turns out that we are firing too many nodes - 1 (consuming node) + 6 (stimulating consuming nodes) + 6 (serving nodes).

If we can manage with a lower number of nodes, it'll be better. 😉